// gatsby-interface/dist/src/components/VisuallyHidden/VisuallyHidden.esm.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import { jsx } from '@emotion/core';
import _pt from 'prop-types';
import { visuallyHiddenCss } from '../../stylesheets/a11y.esm.js';

function VisuallyHidden(_ref) {
  var children = _ref.children,
      rest = _objectWithoutProperties(_ref, ["children"]);

  return jsx("span", _extends({
    css: visuallyHiddenCss
  }, rest), children);
}
VisuallyHidden.propTypes = {
  children: _pt.node.isRequired
};
VisuallyHidden.__docgenInfo = {
  "description": "",
  "methods": [],
  "displayName": "VisuallyHidden",
  "props": {
    "children": {
      "required": true,
      "tsType": {
        "name": "ReactReactNode",
        "raw": "React.ReactNode"
      },
      "description": ""
    }
  }
};

if (typeof STORYBOOK_REACT_CLASSES !== "undefined") {
  STORYBOOK_REACT_CLASSES["src/components/VisuallyHidden/VisuallyHidden.tsx"] = {
    name: "VisuallyHidden",
    docgenInfo: VisuallyHidden.__docgenInfo,
    path: "src/components/VisuallyHidden/VisuallyHidden.tsx"
  };
}

export { VisuallyHidden };
